pkg/build/controller/controller.go

```diff
@@ -79,21 +81,74 @@ func (bc *BuildController) CancelBuild(build *buildapi.Build) error {
 // HandleBuild deletes pods for cancelled builds and takes new builds and puts
 // them in the pending state after creating a corresponding pod
 func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
-	glog.V(4).Infof("Handling build %s/%s", build.Namespace, build.Name)
+	glog.V(4).Infof("Handling build %s/%s (%s)", build.Namespace, build.Name, build.Status.Phase)
+
+	// If we receive build that reached its final state and the build was created
+	// from a serial build configuration, tag the next build in the queue for
+	// execution. That will produce event and the build should get executed
+	// immediately.
```

Comment:
don't you also need to do this if the build that just completed was a parallel build (as long as no other parallel builds are running).